### ISSUE TITLE
Question: is it possible to use it on Android device?

### ISSUE THREAD

**Issue by AndroidDeveloperLB** [DmitryUlyanov/deep-image-prior#37]
does it take a lot of computing power to use this ?
Does it require training ? of a lot of images? 
Can it have the training data built into the app?  Without any server?

**Comment by ZurMaD** [comment]
> does it take a lot of computing power to use this ?
> Does it require training ? of a lot of images?
> Can it have the training data built into the app? Without any server?

Yes, It's a lot of GPU

**Comment by AndroidDeveloperLB** [comment]
So, supposed I have 1000 images of 4K quality, how much time could it take to train on PC vs one of the newest smartphones out there over the past year (say, Pixel 2) ?

**Comment by IEWbgfnYDwHRoRRSKtkdyMDUzgdwuBYgDKtDJWd** [comment]
dude, it will take literally a million years. stop asking stupid questions and learn about the basics.

**Comment by AndroidDeveloperLB** [comment]
@IEWbgfnYDwHRoRRSKtkdyMDUzgdwuBYgDKtDJWd It's not stupid.
Smartphone can do long work. Not at the same level as PC, but still...

**Comment by ZurMaD** [comment]
> So, supposed I have 1000 images of 4K quality, how much time could it take to train on PC vs one of the newest smartphones out there over the past year (say, Pixel 2) ?

GPU of PC is the reason why PC can works normaly with this kind of processing. However, The smartphones are designed to just look nice.

**Comment by AndroidDeveloperLB** [comment]
@ZurMaD Do you think that one day, smarphones will reach the same performance as PC (on the same time era) ? 
I think they are becoming more and more powerful, faster than PCs, so maybe it is possible.

**Comment by ZurMaD** [comment]
> @ZurMaD Do you think that one day, smarphones will reach the same performance as PC (on the same time era) ?
> I think they are becoming more and more powerful, faster than PCs, so maybe it is possible.
> I think it was said a year ago (or so) by ARM, that smartphones have reached a level comparable to consoles, in some way. Is it true?

In reality, large companies are building processes in the cloud, making smartphones more independent with the Internet, because they can optimize. As Facebook Face Detector in the app.

**Comment by ZurMaD** [comment]
> @ZurMaD I see, but what is Tensor Lite, then?
> It's not meant for machine learning, right on the smartphone?
> Suppose you do all the training on very good PCs, can a smartphone have what was learned right on the app, without the need of Internet connection?

Yes, it's totally possible but not recommended because it makes the dispositives too expensive.

**Comment by AndroidDeveloperLB** [comment]
@ZurMaD What do you mean by "dispositives too expensive" ? 
You mean it will take a lot of space in the storage?

**Comment by ZurMaD** [comment]
> @ZurMaD What do you mean by "dispositives too expensive" ?
> You mean it will take a lot of space in the storage?

No, I'm saying that GPUs made devices more expensive.

**Comment by AndroidDeveloperLB** [comment]
Are you talking about PC?
Because smartphones are just what they are... They keep getting replaced by new ones all the time, one small step after another.

**Comment by magneter** [comment]
Actually ,we should evaluate the model to decide wether it is suitable running on the phone 
and the problem we try to solve ,the accuracy ,precision and fps....can not be ignored.
You can optimize  ,remove some branches, reduce the kernels , quantize weights ....
Lots of models have been running on the phone now to sovle problems.
But ,if the problem needs very high precision, and  high fps, especially in efficiency ,you had better run it on the GPUs. For example, this model is  not suitable to be deployed on phone.
